networkmanager/networkmanager_test.go

```diff
@@ -71,542 +99,544 @@ func TestMain(m *testing.M) {
-	}); err != nil {
-		t.Fatalf("Can't add service to network: %s", err)
+	if err := manager.RemoveInstanceFromNetwork(instanceID, "network0"); err != nil {
+		t.Fatalf("Can't remove instance from network: %s", err)
 	}
 
-	if err := runOCIContainer(container1Path, "servicenm3"); err != nil {
-		t.Errorf("Error: %s", err)
+	if _, err := manager.GetInstanceIP(instanceID, "network0"); err == nil {
+		t.Fatal("Instance should not be in network")
 	}
 }
 
-func TestHostName(t *testing.T) {
-	t.Cleanup(func() { _ = manager.DeleteNetwork("network0") })
-
-	container0Path := path.Join(tmpDir, "servicenm4")
-
-	if err := createOCIContainer(container0Path, "servicenm4", []string{"ping", "myhost", "-c10", "-w15"}); err != nil {
-		t.Fatalf("Can't create service container: %s", err)
+func TestFirewallPlugin(t *testing.T) {
+	testData := []testPluginsData{
+		{
+			params: networkmanager.NetworkParams{
+				ExposedPorts:       []string{"900"},
+				AllowedConnections: []string{instanceID + "/900"},
+			},
+			networkConfig: createPlugins([]string{
+				createBridgePlugin(""),
+				createFirewallPlugin("900", "900"),
+				createDNSPlugin(),
+			}),
+		},
+		{
+			params: networkmanager.NetworkParams{
+				ExposedPorts:       []string{"800"},
+				AllowedConnections: []string{instanceID + "/800"},
+			},
+			networkConfig: createPlugins([]string{
+				createBridgePlugin(""),
+				createFirewallPlugin("800", "800"),
+				createDNSPlugin(),
+			}),
+		},
+	}
+	cniConfig := cniConfig{}
+
+	manager, err := networkmanager.New(&config.Config{}, &cniConfig, nil)
+	if err != nil {
+		t.Fatal(err)
 	}
 
-	if err := manager.AddServiceToNetwork("servicenm4", "network0", networkmanager.NetworkParams{
-		Hostname:      "myhost",
-		HostsFilePath: path.Join(container0Path, "etc", "hosts"),
-	}); err != nil {
-		t.Fatalf("Can't add service to network: %s", err)
-	}
+	defer manager.Close()
 
-	if err := runOCIContainer(container0Path, "servicenm4"); err != nil {
-		t.Errorf("Error: %s", err)
-	}
-}
+	networkmanager.GetIPSubnet = getIPSubnet
 
-func TestExposedPortAndAllowedConnection(t *testing.T) {
-	t.Cleanup(func() {
-		_ = killOCIContainer("serviceServer")
-		_ = manager.DeleteNetwork("networkSP1")
-		_ = manager.DeleteNetwork("networkSP2")
-	})
+	for _, item := range testData {
+		if err := manager.AddInstanceToNetwork(instanceID, "network0", item.params); err != nil {
+			t.Fatalf("Can't add instance to network: %s", err)
+		}
 
-	serverPort := "9000"
-	containerServerPath := path.Join(tmpDir, "serviceServer")
-	serverServiceID := "serviceServer"
+		if string(cniConfig.networkConfig.Bytes) != item.networkConfig {
+			t.Errorf("Wrong network config: %s", string(cniConfig.networkConfig.Bytes))
+		}
 
-	if err := createOCIContainerWitHTTPServer(containerServerPath, serverServiceID, serverPort); err != nil {
-		t.Fatalf("Can't create service container: %s", err)
-	}
+		if _, err := manager.GetInstanceIP(instanceID, "network0"); err != nil {
```

Comment:
I guess no need to get instance IP in each test.

Reply:
leave GetInstanceIP only in one test